Rubberduck.Parsing/VBA/ParseCoordinator.cs

```diff
@@ -86,7 +114,60 @@ private void ReparseRequested(object sender, EventArgs e)
                 token = _currentCancellationTokenSource.Token;
             }
 
+            BeginParse(sender,token);
+        }
+
+        public void SuspendRequested(object sender, RubberduckStatusSuspendParserEventArgs e)
+        {
+            object parseRequestor = null;
 
+            try
+            {
+                _parsingSuspendLock.EnterWriteLock();
+                Interlocked.Add(ref _suspensionIteration, 1);
+                var originalStatus = State.Status;
+                _parserStateManager.SetStatusAndFireStateChanged(e.Requestor, ParserState.Busy,
+                    CancellationToken.None);
+                e.BusyAction.Invoke();
+                if (_requestorStack.TryPop(out var lastRequestor))
+                {
+                    _requestorStack.Clear();
+                    parseRequestor = lastRequestor;
+                }
+                else
+                {
+                    if (originalStatus != ParserState.Ready)
+                    {
+                        // We can't assume that we can return to any other state that's not "Ready"; it's better
+                        // to request a state evaluation
+                        _parserStateManager.EvaluateOverallParserState(CancellationToken.None);
+                    }
+                    else
+                    {
+                        _parserStateManager.SetStatusAndFireStateChanged(e.Requestor, originalStatus,
```

Comment:
I agree but in my previous version where I executed a reparse, it looked strange because you'd see state `Inspecting` -> `Busy` -> reparse, and thus we end up doing 2 reparses which seems too jarring from user's POV.

This version would ensure that if we had interrupted the inspections, it would restart at least because it's not the suspender's job to figure where to resume the state, right?

Reply:
I guess you are looking at the label of the refresh button: inspecting is not a state. Have a look at `FindIssuesAsync` in the `Inspector`. At the start, it sets the label to `Inspecting` and the end it sets it to the current status.

Reply:
Indeed. I'm going to try and load several more references to see if I can make the `ResolvngDeclaration` phase run slower so that I might be able to use that to help test manually. If you have another ideas for better manual tests or even unit tests, I'm all eyes.